Enable RFS for font resizing (#29597)
twbs/bootstrap: site/content/docs/4.3/getting-started/rfs.md

---
layout: docs
title: RFS

> description: Bootstrap's resizing engine.

Reviewer: ```suggestion
description: Bootstrap's resizing engine responsively scales common CSS properties to better utilize available space across viewports and devices.
```

group: getting-started
toc: true
---

## What is RFS?

> Bootstrap's side project [RFS](https://github.com/twbs/rfs) is a unit resizing engine which was initially developed to resize font sizes (hence its abbreviation for Responsive Font Sizes). Nowadays RFS is capable of rescaling basically every value for any CSS property with units, like `margin`, `padding`, `border-radius` or even `box-shadow`.

Reviewer: ```suggestion
Bootstrap's side project [RFS](https://github.com/twbs/rfs) is a unit resizing engine which was initially developed to resize font sizes (hence its abbreviation for Responsive Font Sizes). Nowadays RFS is capable of rescaling most CSS properties with unit values like `margin`, `padding`, `border-radius`, or even `box-shadow`.
```

The mechanism automatically calculates the appropriate values based on the dimensions of the browser viewport. It will be compiled into `calc()` functions with a mix of `rem` and viewport units to enable the responsive scaling behavior.

## Using RFS

The mixins are included in Bootstrap and are available once you include Bootstraps' `scss`. RFS can also be installed [standalone](https://github.com/twbs/rfs#installation) if needed.

### Using the mixins

> The `rfs()` mixin has shorthands for `font-size`, `margin`, `margin-top`, `margin-right`, `margin-bottom`, `margin-left`, `padding`, `padding-top`, `padding-right`, `padding-bottom` and `padding-left` which can be used like this:

Reviewer: ```suggestion
The `rfs()` mixin has shorthands for `font-size`, `margin`, `margin-top`, `margin-right`, `margin-bottom`, `margin-left`, `padding`, `padding-top`, `padding-right`, `padding-bottom`, and `padding-left`. See the example below for source Sass and compiled CSS.
```

```scss
.title {
  @include font-size(4rem);
}
```

> which outputs the following CSS:

Reviewer: ```suggestion
```
